framework/src/base/NonlinearSystem.C

```diff
@@ -544,11 +540,16 @@ NonlinearSystem::addKernel(const std::string & kernel_name, const std::string &
     // Create the kernel object via the factory
     MooseSharedPointer<KernelBase> kernel = MooseSharedNamespace::static_pointer_cast<KernelBase>(_factory.create(kernel_name, name, parameters, tid));
 
-    // Extract the SubdomainIDs from the object (via BlockRestrictable class)
-    std::set<SubdomainID> blk_ids = kernel->blockIDs();
-
     // Add the kernel to the warehouse
-    _kernels[tid].addKernel(kernel, blk_ids);
+    _kernels.addObject(kernel, tid);
+
+    // Store time/non-time kernels separately
+    MooseSharedPointer<TimeKernel> t_kernel = MooseSharedNamespace::dynamic_pointer_cast<TimeKernel>(kernel);
+    if (t_kernel)
+      _time_kernels.addObject(kernel, tid);
+    else
```

Comment:
Wait, this doesn't seem right. How do we know that this kernel is either a time kernel or a nodal time kernel. What if it's neither?